### PR TITLE
print <none> when custome column array is out of bound

### LINKED ISSUE #996: custom-columns stops abruptly when hitting a non-existent array index
**What happened**:
When the get pods command with custom-columns option hits an array index that doesn't exist for a pod, the command is aborted:
```
# kubectl get pods -o=custom-columns=STATUS:.status.conditions[1].status
STATUS
True
True
error: array index out of bounds: index 1, length 1
```

**What you expected to happen**:
That the non-existent array element is replaced e.g. by `<none>` in the output. And that the remaining pods are printed:
```
# kubectl get pods -o=custom-columns=STATUS:.status.conditions[1].status
STATUS
True
True
<none>
True
True
True
```

**Environment**:
- Kubernetes client and server versions (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.0-37+7ab055a22f5440", GitCommit:"7ab055a22f5440dbdcd1d41095b7db95d98fc1c3", GitTreeState:"clean", BuildDate:"2020-12-10T18:58:11Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.0-37+7ab055a22f5440", GitCommit:"7ab055a22f5440dbdcd1d41095b7db95d98fc1c3", GitTreeState:"clean", BuildDate:"2020-12-10T18:59:05Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: on-premises microk8s
- OS (e.g: `cat /etc/os-release`): Ubuntu 20.04.1 LTS

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/996

#### Special notes for your reviewer:
custom-columns stops abruptly when hitting a non-existent array index

Before
```
[root@daocloud logs]# /root/kubectl-dev get pods -o=custom-columns=STATUS:.status.conditions[9].status
STATUS
TRUE
error: array index out of bounds: index 9, length 4
TRUE
```

After 

```
[root@daocloud logs]# /root/kubectl-dev get pods -o=custom-columns=STATUS:.status.conditions[9].status
STATUS
TRUE
<none>
TRUE
```

#### Does this PR introduce a user-facing change?
```release-note
kubectl: return none when the custom-column array is out of bound 
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go
Comment (lala123912) [inline staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go:243]: use InfoS  instead. 
klog.V(4).InfoS("Parse failed", "err", err)

Comment (pacoxu) [reply staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go]: Updated. Thanks @lala123912

**Review by soltysh — CHANGES_REQUESTED** [review]
/lgtm cancel
I've tried it locally and it doesn't seem to fully address the problem. Also a unit-test would be nice covering this.

**Comment by pacoxu** [comment]
@soltysh  Thanks for the try.

Let me test locally and add UT for it.

**Comment by dougsland** [comment]
I agree with @soltysh unittest would be nice. @soltysh I have tested locally and it worked for me. Would be nice to get the test you did.

**Comment by dougsland** [comment]
> I agree with @soltysh unittest would be nice. @soltysh I have tested locally and it worked for me. Would be nice to discover
> and cover the failing test you mentioned.

Talked with @soltysh over slack, we will wait the unittest to cover scenarios and I will do a new round of tests :)

**Comment by pacoxu** [comment]
Latest test result:

image

**Comment by pacoxu** [comment]
@dougsland would you take a look again?

**Comment by pacoxu** [comment]
/triage accepted
I think

**Comment by pacoxu** [comment]
Kindly ping @soltysh  @dougsland as UT was added and fix are updated

File: staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go:248]: This is wrong approach, we're loosing information that a user name a mistake, he should be informed about it, but we should continue working on it. So I'd suggest printing error but continue with parsing.

**Review by soltysh — CHANGES_REQUESTED** [review]
Also, please squash into a single commit

File: staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go
Comment (pacoxu) [reply staging/src/k8s.io/kubectl/pkg/cmd/get/customcolumn.go]: Updated.
Print the error and continue with parsing.

**Comment by pacoxu** [comment]
ping @soltysh before code freeze
for approval

**Review by soltysh — CHANGES_REQUESTED** [review]
I looked at it once more and I think we shouldn't be changing this. 
You're basically hiding user error, where he's trying to access a non-existent element in the array. Thus we're giving false information to the user about it's value. 
/hold
/lgtm cancel

**Comment by pacoxu** [comment]
Let me re-think. Only <none> element for “out of index” errors may be better.

**Comment by dougsland** [comment]
I believe we need a fresh review from @soltysh - this new approach seems good. At least give more information about the error and not only None.

**Pull request closed without merging**
